File: app/src/main/java/com/example/kegowidget/MainActivity.java
package com.example.kegowidget;


import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import io.paperdb.Paper;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class MainActivity extends AppCompatActivity {

    Button btnSave, btnRandom;
    EditText  edtTarget, edtTime, edtDate;
    private final String DATE_FORMAT = "HH:mm dd.MM.yyyy";
    long diff_date;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Paper.init(this);

        btnSave = findViewById(R.id.btn_save);
        edtTarget = findViewById(R.id.edt_target);
        edtTime = findViewById(R.id.edt_time);
        edtDate = findViewById(R.id.edt_date);
        btnRandom = findViewById(R.id.rndm_btn);

        btnSave.setOnClickListener(view -> {
            Paper.book().write("target","чтобы " + edtTarget.getText().toString());
            SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_FORMAT);
            try {
                Date event_date = dateFormat.parse(edtTime.getText().toString() + " " + edtDate.getText().toString());
                Date current_date = new Date();
                diff_date = event_date.getTime() - current_date.getTime();

            } catch (ParseException e) {
                e.printStackTrace();
            }
            Paper.book().write("data_and_time", diff_date);
            Toast.makeText(MainActivity.this, "Поехали!",Toast.LENGTH_SHORT).show();
        });
    }
}

File: app/src/main/java/com/example/kego_client/MainActivity.kt
package com.example.kego_client

import android.app.ActivityOptions
import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.util.Pair
import android.view.View
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.textfield.TextInputLayout


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val tasksButton = findViewById<ImageView>(R.id.main_logo)
        tasksButton.setOnClickListener{
            val p1 = Pair.create(tasksButton as View, "logo_transition")
            val options = ActivityOptions
                .makeSceneTransitionAnimation(this@MainActivity, p1)
            val intent = Intent(this@MainActivity, TasksActivity::class.java)
            startActivity(intent, options.toBundle())
        }

        val botInput = findViewById<TextInputLayout>(R.id.bot_input)
        botInput.editText?.addTextChangedListener(object : TextWatcher {

            override fun afterTextChanged(s: Editable) {
                if ('\n' in s.toString()){
                    botInput.editText?.setText("")
                    val p1 = Pair.create(tasksButton as View, "logo_transition")
                    val options = ActivityOptions
                        .makeSceneTransitionAnimation(this@MainActivity, p1)
                    val intent = Intent(this@MainActivity, BotActivity::class.java)
                    startActivity(intent, options.toBundle())
                }
            }

            override fun beforeTextChanged(
                s: CharSequence, start: Int,
                count: Int, after: Int
            ) {
            }

            override fun onTextChanged(
                s: CharSequence, start: Int,
                before: Int, count: Int
            ) {
            }
        })

        val toWidgetConfigButton = findViewById<ImageView>(R.id.to_widget_config)
        toWidgetConfigButton.setOnClickListener{
            val intent = Intent(this@MainActivity, TimerConfigActivity::class.java)
            startActivity(intent)
            overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
        }
    }
}